mcserver/views.py

```diff
@@ -861,24 +1234,23 @@ class TrialViewSet(viewsets.ModelViewSet):
     queryset = Trial.objects.all().order_by("created_at")
     serializer_class = TrialSerializer
 
-    permission_classes = [IsPublic | ((IsOwner | IsAdmin | IsBackend))]
-
+    permission_classes = [IsPublic | (IsOwner | IsAdmin | IsBackend)]
     
     @action(detail=False, permission_classes=[((IsAdmin | IsBackend))])
```

Comment:
for some of the above session views where we were checking permissions within the view, I wonder if we want to instead define permissions for each view like this

Reply:
I am not sure what would be the best approach here. I am not sure about the implications (which way gives us more control over the errors? which one is a standard?)

Reply:
it seems like view-level permissions would be preferable to checking object permission within the view, but we then have less control over the error. It is unlikely that someone's going to get a 'not logged in' error if they are using the system as designed, so I'd kind of prefer the view-level approach.